include/fusilli/node/layernorm_node.h

```diff
@@ -0,0 +1,263 @@
+// Copyright 2025 Advanced Micro Devices, Inc.
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+//===----------------------------------------------------------------------===//
+//
+// This file contains definitions for the layer normalization node
+// `LayernormNode`.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef FUSILLI_NODE_LAYERNORM_NODE_H
+#define FUSILLI_NODE_LAYERNORM_NODE_H
+
+#include "fusilli/attributes/common.h"
+#include "fusilli/attributes/layernorm_attributes.h"
+#include "fusilli/attributes/tensor_attributes.h"
+#include "fusilli/graph/context.h"
+#include "fusilli/node/node.h"
+#include "fusilli/support/logging.h"
+
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace fusilli {
+
+//===----------------------------------------------------------------------===//
+// Layer normalization node.
+//===----------------------------------------------------------------------===//
+
+class LayernormNode : public NodeCRTP<LayernormNode> {
```

Comment:
We also need Layernorm backward at some point (follow-on is fine). It's called `DLNNode` in cudnn: https://sourcegraph.com/github.com/NVIDIA/cudnn-frontend@b372d39879d44c91a8d5b342022e74802b6a8da2/-/blob/include/cudnn_frontend/node/dln.h?L13